common/src/main/java/io/pravega/common/util/StructuredReadableBuffer.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+/**
+ * {@link BufferView} whose contents can be interpreted as structured data.
+ * See {@link StructuredWritableBuffer} for reading such data.
+ */
+public interface StructuredReadableBuffer extends BufferView {
+    /**
+     * Gets the value at the specified index.
+     *
+     * @param index The index to get the value at.
+     * @return The byte at the specified index.
+     * @throws ArrayIndexOutOfBoundsException If index is invalid.
+     */
+    byte get(int index);
+
+    /**
+     * Gets the 16 bit Short value at the specified index.
+     *
+     * @param index The index to get the value at.
+     * @return The Short at the specified index.
+     * @throws ArrayIndexOutOfBoundsException If index is invalid or if there is insufficient space in the array starting
+     *                                        at the specified index to fit the given value.
+     */
+    short getShort(int index);
+
+    /**
+     * Gets the 32 bit Int value at the specified index.
+     *
+     * @param index The index to get the value at.
+     * @return The Integer at the specified index.
+     * @throws ArrayIndexOutOfBoundsException If index is invalid or if there is insufficient space in the array starting
+     *                                        at the specified index to fit the given value.
+     */
+    int getInt(int index);
+
+    /**
+     * Gets the 64 bit Long value at the specified index.
+     *
+     * @param index The index to get the value at.
+     * @return The Long at the specified index.
+     * @throws ArrayIndexOutOfBoundsException If index is invalid or if there is insufficient space in the array starting
+     *                                        at the specified index to fit the given value.
+     */
+    long getLong(int index);
+
+    /**
+     * Gets a 64 bit Unsigned Long from the specified index. This value must have been serialized using
+     * {@link StructuredWritableBuffer#setUnsignedLong} for proper results. This method is not interoperable with
+     * {@link StructuredWritableBuffer#setLong}.
+     *
+     * @param index The index to get the value at.
+     * @return The Long at the specified index.
+     * @throws ArrayIndexOutOfBoundsException If index is invalid or if there is insufficient space in the array starting
+     *                                        at the specified index to fit the given value.
+     */
+    default long getUnsignedLong(int index) {
```

Comment:
It's very odd that this is called "unsigned long" because, as I understand it, the point of this method is so that negative numbers are sorted below positive ones when sorting by bytes. So it's not only relevant when dealing with exclusively positive numbers, if that were the case the method wouldn't be needed in the first place. 
I would term this "sortable" rather than unsigned, and I wouldn't make it part of this interface but a wrapper that implements all these same `getInt` `getShort` etc, so that it works with all of them.

Reply:
I moved this method from `BitConverter` while preserving the name. 

We can consider renaming it in a subsequent change.